pkg/controller/middleware/auth.go

```diff
@@ -77,84 +82,41 @@ func (h *RequireRealmAdminHandler) Handle(next http.Handler) http.Handler {
 		}(); err != nil {
 			h.logger.Errorw("RequireRealmAdmin", "error", err)
 
-			if errors.Is(err, ErrNotRealmAdmin) {
-				flash.FromContext(w, r).Error("You are not authorized to admin that realm.")
-				http.Redirect(w, r, "/realm", http.StatusFound)
+			if controller.IsJSONContentType(r) {
+				h.h.RenderJSON(w, http.StatusUnauthorized, nil)
 				return
 			}
 
 			flash.FromContext(w, r).Error("Unauthorized")
 			http.Redirect(w, r, "/signout", http.StatusFound)
 			return
 		}
-		next.ServeHTTP(w, r)
-	})
-}
-
-type RequireAdminHandler struct {
```

Comment:
This was replaced with realmadmin

Reply:
We still need the require admin handler for the ability to create/delete realms - can I haz back?

Reply:
Since you asked nicely...